index.js

```diff
@@ -0,0 +1,171 @@
+    }
+
+    let groupsOfThrees = numberToGroupsOfThrees(numberToConvert);
+    let hundreds = groupsOfThrees[0];
+    let thousands = groupsOfThrees[1];
+    
+    let wordifiedHundreds = convertToWords(hundreds);
+    let convertedNumber = wordifiedHundreds;
+    let wordifiedThousands = [];
+    
+    if(thousands && thousands == 1) {
+        hundreds == "000" ? null : wordifiedThousands.push("e");
+        wordifiedThousands.push("mil");
+        convertedNumber = wordifiedHundreds.concat(...wordifiedThousands);
+            
+    } else if (thousands) {
+        hundreds == "000" ? null : wordifiedThousands.push("e");
+        wordifiedThousands.push("mil");
+        wordifiedThousands.push(convertToWords(thousands));
+        convertedNumber = wordifiedHundreds.concat(...wordifiedThousands);
+    }
+
+    if(isNegative) {
+        convertedNumber.push("menos");
+    }
+
+    return convertedNumber.reverse().join(' ').trim();
+}
+
+function convertToWords(hundreds) {
+
+    hundreds = stringSpliter(hundreds).reverse();
+    let unit = hundreds[0]; 
+    let ten = hundreds[1];
+    let hundred = hundreds[2];
+
+    let wordifiedHundreds = [];
+
+    if (ten == 1) {
+        wordifiedHundreds.push(uniqueTens[ten + unit]);
+    } else {            
+        wordifiedHundreds.push(numbersEnum[unit].units);
+        if (ten && ten != 0) {
+            unit == 0 ? null : wordifiedHundreds.push("e");
+            wordifiedHundreds.push(numbersEnum[ten].tens);
+        }
+    }
+
+    if (hundred == 1 && ( unit == 0 && ten == 0)) {
+        wordifiedHundreds.push("cem");
+    } else if(hundred & hundred != 0) {
+        (unit == 0 && ten == 0)? null : wordifiedHundreds.push("e");
+        wordifiedHundreds.push(numbersEnum[hundred].hundreds);
+    }
+
+    return wordifiedHundreds;
+}
+
+function numberToGroupsOfThrees(numberToConvert) {
+    let groupsOfThrees = [];
+    let start = numberToConvert.length;
+    let end = 0;
+    while(start > 0) {
+        end = start;
+        groupsOfThrees.push(numberToConvert.slice((Math.max(0, start-3)), end));
+        start = Math.max(0, start-3);
+    }
+    return groupsOfThrees;
+}
+
+function parameterCleaner(receivedParameter) {
+    return "" + parseInt(receivedParameter);
+}
+
+function validate(receivedParameter) {
+    
+    if(!isNumber(receivedParameter)) { 
+        throw new Error("Um número era esperado, ele deve tem um valor entre -99999 e 99999.");
+    }
+
+    if(!isCompatibleSize(receivedParameter)) {
+        throw new Error("Seu número tem um valor fora dos padrões, ele deve ter um valor entre -99999 e 99999.");
+    }
+
+    if(!Number.isInteger(Number(receivedParameter))) {
+        throw new Error("Seu número deve ser inteiro, ele deve ter um valor entre -99999 e 99999.");
+    }
+    return true;
+}
+
+function isNumber(receivedParameter) {
+    if(receivedParameter == "") {
+        return false;
+    }
+    return !isNaN(receivedParameter);
```

Comment:
vários drawbacks em fazer essa validação assim

https://stackoverflow.com/questions/18082/validate-decimal-numbers-in-javascript-isnumeric

Reply:
Optei por utilizar `return !isNaN(parseFloat(n)) && isFinite(n);`  devido aos testes unitários providos pelo usuário CMS.